src/Components/MainSection.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from 'react'
+import gif from '../Images/loading.gif'
+
+
+
+const MainSection = () => {
+    // loading state  https://swapi.dev/api/films
+    const [loading, setLoading] = useState(true);
+
+    // data state
+    const [data, setData] = useState(null);
+
+    // error state
+    const [error, setError] = useState(null);
+
+
+    // fetching the data from an API
+    useEffect(() => {
+        fetch('https://swapi.dev/api/films')
+            .then((response) => {
+                if (!response.ok) {
+                    alert('This is an HTTP Error: The status is ${response.status}');
+                    throw new Error('This is an HTTP Error: The status is ${response.status}');
+                }
+                return response.json();
+            })
+
+            .then((actualData) => {
+                setData(actualData.results)
+                setError(null)
+
+            })
+            .catch((error) => {
+                console.log(error);
+                setError(error)
+                setData(null)
+            })
+            .finally(() => {
+                setLoading(false);
+            })
+
+    }, [])
+
+
+    return (
+        <div>
+            {loading && <div><img className='loading' src={gif} alt="loading gif" /></div>}
+            {error && <div>{'There is a problem fetching your data - ${error}'}</div>}
```

Comment:
you are meant to use template strings here since you have an error variable. So the right way to so that would be to replace the quotes with back ticks (`)